chain/x/mint/genesis.go

```diff
@@ -1,18 +1,36 @@
 package mint
 
 import (
+	"fmt"
+	"github.com/GeoDB-Limited/odincore/chain/x/mint/internal/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
 // InitGenesis new mint genesis
-func InitGenesis(ctx sdk.Context, keeper Keeper, data GenesisState) {
+func InitGenesis(ctx sdk.Context, keeper Keeper, supplyKeeper types.SupplyKeeper, data GenesisState) {
```

Comment:
do not change the interface of this func, use needed logic from inner keeper (maybe add some proxy functions to call inner supplyKeeper from keeper)